test/test_lexer.ml

```diff
@@ -0,0 +1,50 @@
+open Ts_compiler
+open OUnit2
+
+let rec collect l acc =
+  match Lexer.next_token l with
+  | _, EOF -> List.rev (Token.EOF :: acc)
+  | l', token -> collect l' (token :: acc)
+
+let collect_tokens lexer = collect lexer []
+
+let make_next_token_test name program expected =
+  let lexer = Lexer.of_string program in
+  name >:: fun _ ->
+  assert_equal expected (collect_tokens lexer)
+    ~printer:Helpers.(string_of_list Lexer.show_token)
+
+let next_token_tests =
+  "next_token tests"
+  >::: [
+         make_next_token_test "should handle an empty program" "" [ EOF ];
+         make_next_token_test "should handle program that is just whitespace"
+           "   \n\t\r\n" [ EOF ];
+         make_next_token_test "should correctly scan an assignment expression"
+           "const x = 1 + 2;"
+           [ Const; Iden "x"; Equal; Int 1; Add; Int 2; Semi; EOF ];
+         make_next_token_test
+           "should correctly scan a compact assignment expression"
+           "const variable=1+2;"
+           [ Const; Iden "variable"; Equal; Int 1; Add; Int 2; Semi; EOF ];
+       ]
+
+let peek_token_tests =
+  "peek_token tests"
```

Comment:
This is sort of a dumb tests, IMO. Since our lexer is inherently immutable, the fact that `peek_token` does not return a lexer makes this useless and can never fail.

I wrote it though just in case we decide to make it mutable 🤷